include/RED4ext/SharedPtr.hpp

```diff
@@ -0,0 +1,377 @@
+#pragma once
```

Comment:
Can you move this to Memory dir? I would like to start having these more organize, looking at `DynArray`, `SortedArray` and friends.